src/codegen/riscv64/constants-riscv64.h

```diff
@@ -978,6 +978,36 @@ class InstructionGetters : public T {
     return imm6 << 26 >> 26;
   }
 
+  inline int RvcImm6UValue() const {
+    DCHECK(this->IsShortInstruction());
+    // | funct3 | nzimm[17] | rs1/rd | nzimm[16:12] | opcode |
+    //  15         12                 6            2        
+    int32_t imm6 = this->RvcImm6Value();
+    DCHECK(imm6 != 0);
+    return (uint32_t)imm6 << 12 >> 12;
+  }
+  
+  inline int RvcImm6Addi16spValue() const {
+    DCHECK(this->IsShortInstruction());
+    // | funct3 | nzimm[9] | 2 | nzimm[4|6|8:7|5] | opcode |
+    //  15         12           6                2        
+    uint32_t Bits = this->InstructionBits();
+    int32_t imm6 = ((Bits & 0x1000) >> 3) | ((Bits & 0x40) >> 2) 
+                  | ((Bits & 0x20) << 1) | ((Bits & 0x18) << 4) 
+                  | ((Bits & 0x4) << 3);
+    DCHECK(imm6 != 0);
+    return imm6 << 22 >> 22;
+  }
+
+  inline int RvcShamt() const {
+    DCHECK(this->IsShortInstruction());
+    // | funct3 | nzuimm[5] | rs1/rd | nzuimm[4:0] | opcode |
+    //  15         12                 6           2        
+    int32_t imm6 = this->RvcImm6Value();
+    DCHECK(imm6 != 0);
```

Comment:
According to the spec

> C.SLLI is a CI-format instruction that performs a logical left shift of the value in register rd then
writes the result to rd. The shift amount is encoded in the shamt field. For RV128C, a shift amount
of zero is used to encode a shift of 64. C.SLLI expands into slli rd, rd, shamt[5:0], except for
RV128C with shamt=0, which expands to slli rd, rd, 64.

A shift amount of 0 is allowed but interpreted as shifting by 64. So no need for `DECH(imm6 != 0)` here. But the simulator needs to handle the shift-amount of 0 as a special case.

Reply:
RV128C means that the base ISAs are RV128. So we should ignore RV128C and just implement RV64C.

Reply:
What I meant is that for `RV64C`, the shift-amount can be zero, so `DCHECK(imm6 != 0)` is too restrictive.